sql/ddl_api.sql

```diff
@@ -81,6 +84,31 @@ BEGIN
         USING ERRCODE = 'IO102';
     END IF;
 
+    -- We don't use INTO STRICT here because that error (no column) is surfaced later.
+    SELECT atttypid
+    INTO time_type
+    FROM pg_attribute
+    WHERE attrelid = main_table AND attname = time_column_name;
+
+    -- Timestamp types can use default value, integral should be an error if NULL
+    IF time_type IN ('TIMESTAMP', 'TIMESTAMPTZ') AND chunk_time_interval IS NULL THEN
+        chunk_time_interval_actual := _timescaledb_internal.interval_to_usec('1 month');
+    ELSIF time_type IN ('SMALLINT', 'INTEGER', 'BIGINT') AND chunk_time_interval IS NULL THEN
```

Comment:
should we consider supporting integer types like SERIAL, and BIGSERIAL. And, what about decimal types? Or, should such additions be a separate PR?